### PR TITLE
bin.strpurge: added support to prevent string purge by prev entries (addr range)

### PR DESCRIPTION
**Pull request by kazarmy**

### CODE REVIEW

File: libr/core/cbin.c [hunk 1]
```diff
@@ -215,21 +215,21 @@ R_API bool r_core_bin_strpurge(RCore *core, const char *str, ut64 refaddr) {
...
+				}
 				range_sep = strchr (ptr, '-');
 				if (range_sep) {
 					*range_sep = 0;
 					from = r_num_get (NULL, ptr);
 					ptr = range_sep + 1;
 					to = r_num_get (NULL, ptr);
 					if (refaddr >= from && refaddr <= to) {
-						purge = true;
+						purge = bang ? false : true;
```
Comment (radare) [inline libr/core/cbin.c]: You can use just !bang

File: libr/core/cconfig.c [hunk 2]
```diff
@@ -2391,10 +2391,11 @@ R_API int r_core_config_init(RCore *core) {
 	SETOPTIONS (n, "latin1", "utf8", "utf16le", "utf32le", "guess", NULL);
 	SETCB ("bin.usextr", "true", &cb_usextr, "Use extract plugins when loading files");
 	SETCB ("bin.useldr", "true", &cb_useldr, "Use loader plugins when loading files");
-	SETCB ("bin.strpurge", "", &cb_strpurge, "Try to purge false positive strings (true: use the "
-	       "false_positive() classifier in cbin.c, [,addr]*: specific string addrs to purge, "
-	       "[,addr1-addr2]*: purge all strings in the range addr1-addr2 inclusive, [,!addr]*: prevent "
-	       "purge of string at addr by prev entries)");
+	SETCB ("bin.strpurge", "", &cb_strpurge, "Purge strings (true: use the false_positive() "
```
Comment (radare) [inline libr/core/cconfig.c]: Seems too long description to me

File: libr/core/cbin.c (hunk 1 above)
Comment (kazarmy) [reply libr/core/cbin.c]: fixed in 7c8252c6f3e5ddcd3a2d3f4a592fb6e2397b4550

File: libr/core/cconfig.c (hunk 2 above)
Comment (kazarmy) [reply libr/core/cconfig.c]: see #9232

**Pull request merged**